src/components/SignUpForm/index.tsx

```diff
@@ -0,0 +1,318 @@
+import { useState, useCallback, SyntheticEvent } from 'react';
+import { Link } from 'react-router-dom';
+import {
+  ID_REQUIRE_CHECK,
+  NICKNAME_REQUIRE_CHECK,
+  PW_REQUIRE_CHECK,
+  PW_VALID_CHECK,
+  PWCONFIRM_REQUIRE_CHECK,
+  PWCONFIRM_VALID_CHECK,
+  EMAIL_REQUIRE_CHECK,
+  EMAIL_VALID_CHECK,
+  MBTI_REQUIRE_CHECK,
+} from '../../constants/message';
+import { regEmail } from '../../constants/regEx';
+import LabelBasicInput from '../LabelBasicInput';
+import LabelBasicSelect from '../LabelBasicSelect';
+import Button from '../styled-components/Button';
+import './style.scss';
+
+// Mbti select option interface
+interface Mbti {
+  id: string;
+  name: string;
+}
+
+// Gender select option interface
+interface Gender {
+  id: string;
+  name: string;
+}
+
+const SignUpForm = () => {
+  const [loginId, setLoginId] = useState<string>('');
+  const [nickname, setNickname] = useState<string>('');
+  const [password, setPassword] = useState<string>('');
+  const [passwordConfirm, setPasswordConfirm] = useState<string>('');
+  const [email, setEmail] = useState<string>('');
+  const [mbti, setMbti] = useState<string>('1');
+  const [gender, setGender] = useState<string>('1');
+  // Valid
+  const [isValidLoginId, setIsValidLoginId] = useState<boolean>(false);
+  const [isValidNickname, setIsValidNickname] = useState<boolean>(false);
+  const [isValidPassword, setIsValidPassword] = useState<boolean>(false);
+  const [isValidPasswordConfirm, setIsValidPasswordConfirm] = useState<boolean>(false);
+  const [isValidEmail, setIsValidEmail] = useState<boolean>(false);
+  const [isValidMbti, setIsValidMbti] = useState<boolean>(false);
+  // Error Message
+  const [loginIdErrorMessage, setLoginIdErrorMessage] = useState<string>(ID_REQUIRE_CHECK); // 추후 아이디 중복 체크시 에러메시지 useState
+  const [nicknameErrorMessage, setNickNameErrorMessage] = useState<string>(NICKNAME_REQUIRE_CHECK);
+  const [passwordErrorMessage, setPasswordErrorMessage] = useState<string>(PW_REQUIRE_CHECK);
+  const [passwordConfirmErrorMessage, setPasswordConfirmErrorMessage] =
+    useState<string>(PWCONFIRM_REQUIRE_CHECK);
+  const [emailErrorMessage, setEmailErrorMessage] = useState<string>(EMAIL_REQUIRE_CHECK);
+  const [mbtiErrorMessage, setMbtiErrorMessage] = useState<string>(MBTI_REQUIRE_CHECK);
```

Comment:
```suggestion
  const [validationState, setValidationState] = useState<SignUpValidationState>({
    loginId: { isValid: false, errorMessage: '아이디를 입력해주세요.' },
    nickname: { isValid: false, errorMessage: '닉네임을 입력해주세요.' },
    password: { isValid: false, errorMessage: '비밀번호를 입력해주세요.' },
    passwordConfirm: { isValid: false, errorMessage: '비밀번호를 재입력해주세요.' },
    email: { isValid: false, errorMessage: '이메일을 입력해주세요.' },
    mbti: { isValid: false, errorMessage: 'MBTI를 선택해주세요.' },
  });
```
이렇게 바꾼다면 유틸 파일에서 많은 상수 변수들을 import안해도 같은 결과를 낼 수 있을 거같아요.